pkg/ruler/api.go

```diff
@@ -163,19 +117,19 @@ func (a *API) PrometheusRules(w http.ResponseWriter, req *http.Request) {
 	userID, err := tenant.TenantID(req.Context())
 	if err != nil || userID == "" {
 		level.Error(logger).Log("msg", "error extracting org id from context", "err", err)
-		respondError(logger, w, "no valid org id found")
```

Comment:
I think if you run cortex without an auth service in front  this change would make sense because its the client's fault for not setting the header, but if there is another service that sets the header then we will be returning 4xx when that service is not working fine. My vote is keeping the 5xx just because I think its more common that it's another service that sets the header and not the client itself.

Reply:
When one Ruler calls other Rulers to fetch rules for example, those are gRPC calls and does not go thru API layer. It calls `Rules(ctx context.Context, in *RulesRequest)`. So it is OK I think